p2p/rlp.py

```diff
@@ -0,0 +1,33 @@
+import rlp
+from rlp import sedes
+
+from evm.rlp.headers import BlockHeader
+from evm.rlp.transactions import BaseTransaction
+
+
+# This is needed because BaseTransaction has several @abstractmethods, which means it can't be
+# instantiated.
+class P2PTransaction(BaseTransaction):
+
+    def check_signature_validity(self):
```

Comment:
Event though these methods doesn't do much I think we should annotate them as `-> None`. I hope that once the entire code base is annotated we will enable checks to require all code to be annotated hence I think we should already try our best to land new code always with annotations.

Reply:
We'll never implement those methods here, so I don't see the point. Type hints are specially useful when you have non-trivial/non-obvious things -- there's even plenty of people who argue there's no need for type hints when an argument's type is obvious.

But yeah, we'll definitely want to enable as many mypy checks as possible, eventually.

Reply:
>We'll never implement those methods here, so I don't see the point

It's about consistency. At some point we could enable `--disallow-untyped-defs` in `mypy` which would basically reject any function to land that isn't typed (with the exception of cases that may need a `type: ignore` because they are too dynamic to type out).

>there's even plenty of people who argue there's no need for type hints when an argument's type is obvious

I see the point, I guess I just value consistency more. Something may be obvious to someone but isn't obvious to someone else so I'd rather stay consistent and just require all function definitions to be typed.